src/runtimes/node/utils/tsconfig.ts

```diff
@@ -0,0 +1,19 @@
+import { getTsconfig } from 'get-tsconfig'
+
+import { MODULE_FORMAT } from './module_format.js'
+
+const esmModuleValues = new Set(['es2020', 'es2022', 'esnext', 'node16', 'nodenext'])
+
+// Returns the module format that should be used for a TypeScript file at a
+// given path, by reading the associated `tsconfig.json` file if it exists.
+export const getModuleFormat = (path: string) => {
+  const file = getTsconfig(path)
+
+  if (!file) {
+    return MODULE_FORMAT.COMMONJS
+  }
+
+  const moduleProp = file.config.compilerOptions?.module?.toLowerCase() ?? ''
+
+  return esmModuleValues.has(moduleProp) ? MODULE_FORMAT.ESM : MODULE_FORMAT.COMMONJS
```

Comment:
According to https://www.typescriptlang.org/tsconfig#module, it's a bit more complicated I think:

<img width="257" alt="Screenshot 2023-08-02 at 15 43 59" src="https://github.com/netlify/zip-it-and-ship-it/assets/14912729/a1c69cd4-9940-4b81-84c0-79222a777163">

It looks like we also need to emit ESM for `es6` and `es2015`. Maybe it's easier to phrase this in terms of when we need to emit CJS?

Reply:
yea if `module` is es2015 then it results in esm as well:
https://www.typescriptlang.org/play?target=99&moduleResolution=99&module=5&jsx=0#code/KYDwDg9gTgLgBAMwK4DsDGMCWEVwBbAA2hEAFAJRwDeAUHHGjgM4SHAB0JA5qZgLaRY7PsBgBDdkiiFyNAL5A
![CleanShot 2023-08-02 at 17 16 32](https://github.com/netlify/zip-it-and-ship-it/assets/11156362/356e700b-21ff-4beb-b15b-29cf60a3a563)

Reply:
Done in https://github.com/netlify/zip-it-and-ship-it/pull/1505/commits/217633cb74db61f145105a820d16a91405eb3692.